tests/t_ethereum_verkle_ipa_primitives.nim

```diff
@@ -276,46 +278,115 @@ suite "Transcript Tests":
 
-      doAssert (challenge1 == challenge2).bool() == false , "calling ChallengeScalar twice should yield two different challenges"
+      # Comparing with Geth implementation
+      doAssert b1.toHex() == "0xc2aa02607cbdf5595f00ee0dd94a2bbff0bed6a2bf8452ada9011eadb538d003", "Incorrect Value!"
 
     testVec()
 
   test "Transcript testing with same challenge scalar to test transcript correctness":
 
     proc testVec1()=
 
+      # Initializing 2 new transcript states
       var tr {.noInit.}: sha256
       var tr2 {.noInit.}: sha256
+
+      # Generating 2 new labels into 2 separate transcripts
       tr.newTranscriptGen(asBytes"simple_protocol")
       tr2.newTranscriptGen(asBytes"simple_protocol")
       
-
+      # Generating Challenge Scalar for Transcript 1
       var challenge1 {.noInit.}: matchingOrderBigInt(Banderwagon)
       challenge1.generateChallengeScalar(tr,asBytes"ethereum_challenge")
 
+      # Generating Challenge Scalar for Transcript 2
       var challenge2 {.noInit.}: matchingOrderBigInt(Banderwagon)
       challenge2.generateChallengeScalar(tr2,asBytes"ethereum_challenge")
 
+      # Challenge 1 should be equal to Challenge 2 as both are coming from different transcript
+      # states that are being handled similarly
       doAssert (challenge1 == challenge2).bool() == true , "calling ChallengeScalar twice should yield the same challenge"
 
     testVec1()
 
+  test "Transcript testing with repititive append of scalars, thereby a compound challenge scalar":
+    proc testVec2()=
+
+      # Initializing a new transcript state
+      var tr {.noInit.}: sha256
+
+      # Generating with a new label
+      tr.newTranscriptGen(asBytes"simple_protocol")
+
+      var five {.noInit.} : matchingOrderBigInt(Banderwagon)
+      five.fromUint(uint64(5))
+
+      # Appending some scalars to the transcript state
+      tr.scalarAppend(asBytes"five", five)
+      tr.scalarAppend(asBytes"five again", five)
+
+      var challenge {.noInit.}: matchingOrderBigInt(Banderwagon)
+      challenge.generateChallengeScalar(tr, asBytes"simple_challenge")
+
+      var c_bytes {.noInit.}: array[32, byte]
+      discard c_bytes.serialize_scalar(challenge, littleEndian)
+      
+      # Comparing with Geth Implmentation
+      doAssert c_bytes.toHex() == "0x498732b694a8ae1622d4a9347535be589e4aee6999ffc0181d13fe9e4d037b0b", "Some issue in Challenge Scalar"
+    
+    testVec2()
+
+    test "Transcript testing with +1 and -1, appending them to be a compound challenge scalar":
+      proc testVec3() = 
+
+        # Initializing a new transcript state
+        var tr {.noInit.}: sha256
+  
+        # Generating with a new label
+        tr.newTranscriptGen(asBytes"simple_protocol")
+
+        var one {.noInit.}: matchingOrderBigInt(Banderwagon)
+        var minus_one {.noInit.}: Fr[Banderwagon]
+
+        minus_one.setMinusOne()
+        one.setOne()
+
+        # Constructing a Compound Challenge Scalar
```

Comment:
Explain a bit more about this operations in comments

Reply:
sure

Reply:
done @advaita-saha